tests/conftest.py

```diff
@@ -76,13 +108,9 @@ def docker_client():
 
 
 def pytest_addoption(parser):
-    parser.addoption(
-        "--armory-mode",
-        action="store",
-        default="docker",
-        choices=["native", "docker", "both"],
-        help="Set Armory Mode [native|docker|both]",
-    )
+    for args, kwargs in armory_test_parameters:
+        args = args if isinstance(args, tuple) else (args,)
```

Comment:
Avoid polymorphic arguments without necessity.  Since you are writing the contents of `armory_test_parameters` you can ensure they are properly typed. This `isinstance` encourages slop.